File: src/main/java/org/bag/domain/Image.java
package org.bag.domain;

import javax.persistence.*;
import java.util.Objects;

/**
 * Plaatje behorende bij een project.
 */
@Entity
public class Image {
    @Id
    @GeneratedValue(strategy = GenerationType.AUTO)
    private int id;
    private String imageUrl;

    protected Image(){}

    public Image(String imageUrl) {
        this.imageUrl = imageUrl;
    }

    public int id() {
        return id;
    }

    public String url() {
        return imageUrl;
    }

    @Override
    public boolean equals(Object o) {
        if (this == o) return true;
        if (o == null || getClass() != o.getClass()) return false;
        Image that = (Image) o;
        return imageUrl.equals(that.imageUrl);
    }

    @Override
    public int hashCode() {
        return Objects.hash(imageUrl);
    }
}

File: website/src/main/kotlin/org/bag/domain/Image.kt
package org.bag.domain

import javax.persistence.*
import java.util.Objects

/**
 * Plaatje behorende bij een project.
 */
@Entity
class Image {
    @Id
    @GeneratedValue(strategy = GenerationType.AUTO)
    val id: Int = 0

    val url: String

    constructor(imageUrl: String) {
        this.url = imageUrl
    }

    override fun equals(o: Any?): Boolean {
        if (this === o) return true
        val that = o as Image?
        return url == that!!.url
    }

    override fun hashCode(): Int {
        return Objects.hash(url)
    }

    private constructor(){
        this.url = ""
    }
}
